test.sh

```diff
@@ -0,0 +1,254 @@
+#!/bin/bash
+set -e -o pipefail
+
+help() {
+    echo 'Usage ./test.sh [-f docker-compose.yml] [-p project] [args]'
+    echo
+    echo 'Optional args'
+    echo '  run:            [default] starts up the entire stack.'
+    echo '  check:          verify your local environment is correctly configured.'
+    echo '  show:           open web pages of an already running stack.'
+    echo '  test <logtype>: run test client against an already running stack. logtype'
+    echo '                  should be one of: syslog, gelf'
+    echo '  help            help. you are reading it now.'
+    echo
+    echo 'Optional flags:'
+    echo '  -f <filename>   use this file as the docker-compose config file'
+    echo '  -p <project>    use this name as the project prefix for docker-compose'
+    echo
+}
+
+
+# default values which can be overriden by -f or -p flags
+export COMPOSE_PROJECT_NAME=elk
+export COMPOSE_FILE=
+
+# give the docker remote api more time before timeout
+export COMPOSE_HTTP_TIMEOUT=300
+
+# populated by `check` function whenever we're using Triton
+TRITON_USER=
+TRITON_DC=
+TRITON_ACCOUNT=
+
+# ---------------------------------------------------
+# Top-level commmands
+
+run() {
+    check
+    docker-compose up -d
+    show
+}
+
+show() {
+    check
+    # poll Consul for liveness and then open the console
+    poll-for-page "http://$(getPublicUrl consul 8500)/ui/" \
+                  'Waiting for Consul...' \
+                  'Opening Consul console... Refresh the page to watch services register.'
+
+    # poll Kibana for liveness and then open the page
+    poll-for-page "http://$(getPublicUrl kibana 5601)/app/kibana#discover" \
+                  'Waiting for Kibana to register as healthy...' \
+                  'Opening Kibana console.'
+}
+
+# Run test clients
+test() {
+    check
+    local logtype=$1
+    local port
+    local protocol=tcp
+    case $logtype in
+        gelf)
+            port=12201
+            protocol=udp
+            ;;
+        syslog)
+            port=514 ;;
+        # Triton supports fluentd but logstash codec support is broken:
+        # https://github.com/logstash-plugins/logstash-codec-fluent/issues/2
+        # https://github.com/logstash-plugins/logstash-codec-fluent/pull/5
+        # fluentd)
+        #    port=24224 ;;
+        *)
+            echo 'logtype arguments required: gelf or syslog'
+            exit 1;;
+    esac
+
+    echo 'Starting Nginx log source...' && \
+        CONSUL=$(getPrivateIpPort consul 8500 tcp) \
```

Comment:
When running locally we need to use links and not CNS, so the `_env` file which gets used by `docker-compose.yml` is populated by CNS while the `local-compose.yml` file just uses links and the static string "consul". If all we wanted to do was stand up some containers we could just do all this logic in two compose files, but we're doing things like opening the web interfaces.

You've asked about this a couple different ways now, so let me propose this approach:
- we ditch the `test.sh` file entirely
- we provide a `docker-compose.yml`, `local-compose.yml`, `test-compose.yml`, and `test-local-compose.yml` that does all the standing up of things.
- we provide a `check.sh` file to do the environment checking and open web portals, but it only works on Triton and not locally.